staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3075,6 +3075,50 @@ type ContainerStatus struct {
 	// +featureGate=SupplementalGroupsPolicy
 	// +optional
 	User *ContainerUser `json:"user,omitempty" protobuf:"bytes,13,opt,name=user,casttype=ContainerUser"`
+	// AllocatedResourcesStatus represents the status of various resources
+	// allocated for this Pod.
+	// +featureGate=ResourceHealthStatus
+	// +optional
+	AllocatedResourcesStatus ResourcesStatus `json:"allocatedResourcesStatus,omitempty" protobuf:"bytes,14,rep,name=allocatedResourcesStatus,casttype=ResourcesStatus,castkey=ResourceName"`
+}
+
+type ResourcesStatus map[ResourceName]ResourceStatus
+
+type ResourceStatus struct {
+	// map of unique Resource ID to its health.
+	// At a minimum, ResourceID must uniquely identify the Resource
+	// allocated to the Pod on the Node for the lifetime of a Pod.
+	// See ResourceID type for it's definition.
+	Resources map[ResourceID]ResourceHealth `json:"resources,omitempty" protobuf:"bytes,1,rep,name=resources,castkey=ResourceID"`
```

Comment:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps